components/event-bus/internal/knative/util/kn-eventing_test.go

```diff
@@ -90,6 +93,40 @@ func Test_CreateChannel(t *testing.T) {
 	}
 }
 
+func Test_GetChannelByLabels(t *testing.T) {
+	log.Print("Test_GetChannelByLabels")
+	log.Print("Creating Channel to fetch")
+	client := evclientsetfake.NewSimpleClientset()
+	client.Fake.ReactionChain = nil
+	client.Fake.AddReactor("create", "channels", func(action k8stesting.Action) (handled bool,
+		ret runtime.Object, err error) {
+		return true, testChannel, nil
+	})
+	client.Fake.AddReactor("list", "channels", func(action k8stesting.Action) (handled bool,
+		ret runtime.Object, err error) {
+		return true, testChannelList, nil
+	})
+
+	k := &KnativeLib{
+		evClient:         client.EventingV1alpha1(),
+		messagingChannel: client.MessagingV1alpha1(),
+	}
+	ch1, err1 := k.CreateChannel(channelName, testNS, &labels, 10*time.Second)
+	assert.Nil(t, err1)
+	log.Printf("Channel created: %v", ch1)
+	log.Println("Getting Channel By label")
+
+	ch2, err2 := k.GetChannelByLabels(testNS, &labels)
+	assert.Nil(t, err2)
+
+	fmt.Printf("======== %v\n", ch2)
```

Comment:
@anishj0shi Do we really need this print?

Reply:
no..

Reply:
done